bin/pytorch_inference/Main.cc

```diff
@@ -69,11 +69,17 @@ torch::Tensor infer(torch::jit::script::Module& module_,
         }
 
         auto output = module_.forward(inputs);
+        LOG_DEBUG(<< "output_" << i << ": " << output);
         if (output.isTuple()) {
             // For transformers the result tensor is the first element in a tuple.
             all.push_back(output.toTuple()->elements()[0].toTensor());
         } else {
-            all.push_back(output.toTensor());
+            auto outputTensor = output.toTensor();
+            if (outputTensor.dim() == 0) { // If the output is a scaler, we need to reshape it into a 1D tensor
+                all.push_back(outputTensor.reshape({1, 1}));
+            } else {
+                all.push_back(outputTensor);
```

Comment:
Not sure how important it is for performance, but you might be able to use `std::move` here to avoid copying `outputTensor` into the vector. (The way the code was before, the temporary rvalue `output.toTensor()` was added directly to the vector, so would be moved by default. But in the new code it’s stored as an lvalue in a local variable, so cannot be moved unless you specifically ask for this. Maybe it’s not very important. It depends on the internal implementation of the object in PyTorch. If a copy involves a memory allocation then it’s will affect performance much more than if it’s just copying a few builtin types.)

Reply:
Thanks Dave! I hope you're doing well!!

Reply:
I ran the throughput/latency benchmark and the memory benchmark, and the numbers were extremely similar. There might be a slight performance improvement to using `std:move`, so I'll go with that.

Thanks Dave!